### PR TITLE
refactor!: map BigInteger to new BigInt scalar

### LINKED ISSUE #100: Mapping BigInteger to int limits the max value of BigInteger
## Describe the Bug

I have a sqlalchemy model that is set up to something like this:
```py
class UsersModel(Base):
    __tablename__ = "users"

    id = Column(BigInteger, unique=True, primary_key=True)
    # ... other fields
```
and I am mapping the schema like this:
```py

@mapper.type(UsersModel)
class Users:
    ...
```
Some of the IDs that are being used are very large, e.g.: `1_029_320_583_927` which is way beyond the limit of Int32 (`2_147_483_647`), thus when I run a pretty simple query:
```
query {
  users {
    id
    name
  }
}
```
I am getting an error:
`Int cannot represent non 32-bit signed integer value: 1029320583927`

I think the mapping should be done to a custom scalar type in order to allow big integer values.

## System Information

 - Operating system: macOS Sonoma 14.1.2
 - Strawberry version (if applicable): 0.215.1

## Additional Context

Unfortunately the IDs must be BigInteger since this is already being used by other services so I can't change it to be UUID or even String which might be a different approach to solve it.

### PR DESCRIPTION
**Pull request by IdoKendo**
This PR fixes #100 by implementing a new custom scalar `Int64` following the strawberry documentation: https://strawberry.rocks/docs/types/scalars#int64

## Description

The new `Int64` custom scalar resolved issues of very large integers when using SQLAlchemy's `BigInteger`.

## Types of Changes

- [x] Core
- [x] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Documentation

## Issues Fixed or Closed by This PR

* #100 

## Checklist

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the CONTRIBUTING document.
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by erikwrede** [comment]
Thanks for the quick PR! I adjusted the title since this is a breaking change if we're strict. Still want to wait on a second opinion from @mattalbr but for me this is ready to merge. Since we don't have any unit tests on the scalar conversion yet, this is fine for me 😊

**Review by mattalbr — CHANGES_REQUESTED** [review]
Sorry for the slow review! Was tuned out to OSS the past few weeks. I'll approve quickly after this!

File: src/strawberry_sqlalchemy_mapper/scalars.py
Comment (IdoKendo) [reply src/strawberry_sqlalchemy_mapper/scalars.py:5]: Oh, I would much rather pass it as an integer rather than a string, honestly! I was just following the example from:
https://strawberry.rocks/docs/types/scalars#int64
Do you think there's a better way in order to preserve it as an integer?

Comment (mattalbr) [reply src/strawberry_sqlalchemy_mapper/scalars.py:5]: So, you can set the GRAPHQL_MAX_INT, which from our limited testing "works", but the docs recommend doing what you did here creating a new scalar serialized as a string, primarily because javascript only supports up to 53 bits. 

I like what you have here, just slightly prefer the name BigInt to Int64, and possibly also worth including a warning docstring to be careful with these strings in javascript clients.

Comment (IdoKendo) [reply src/strawberry_sqlalchemy_mapper/scalars.py]: I renamed it, do let me know if there's anything else before it can be merged!

**Review by erikwrede — APPROVED** [review]

**Review by mattalbr — APPROVED** [review]

**Comment by botberry** [comment]
Thanks for contributing to Strawberry! 🎉 You've been invited to join
the Strawberry GraphQL organisation 😊

You can also request a free sticker by filling this form: https://forms.gle/dmnfQUPoY5gZbVT67

And don't forget to join our discord server: https://strawberry.rocks/discord 🔥

**Pull request merged**
